src/featurestorebundle/metadata/reader/MetadataReader.py

```diff
@@ -1,11 +1,70 @@
-from typing import Optional
+from typing import List, Optional
+from box import Box
+from pyspark.sql import SparkSession
 from pyspark.sql import DataFrame
 from featurestorebundle.metadata.reader.MetadataReaderInterface import MetadataReaderInterface
 
 
 class MetadataReader:
-    def __init__(self, metadata_reader: MetadataReaderInterface):
+    def __init__(
+        self,
+        stable_feature_store: Box,
+        metadata_reader: MetadataReaderInterface,
+        spark: SparkSession,
+    ):
+        self.__stable_feature_store = stable_feature_store
         self.__metadata_reader = metadata_reader
+        self.__spark = spark
 
     def read(self, entity_name: Optional[str]) -> DataFrame:
         return self.__metadata_reader.read(entity_name)
+
+    def read_for_latest(self, entity_name: Optional[str]) -> DataFrame:
+        if self.__stable_feature_store.metadata.table is None:
+            return self.__read_current_feature_store_metadata(entity_name)
+
+        return self.__read_current_and_stable_feature_store_metadata(entity_name)
+
+    def read_for_target(self, entity_name: Optional[str], target_name: str) -> DataFrame:
+        if (
+            self.__stable_feature_store.metadata.table is None
+            or self.__stable_feature_store.target.table is None
```

Comment:
should this be `and`? if one of them is set and the rest isn't then it should except right?

Reply:
I don't know what if someone will have just prod feature store but not target store